build-logic/build.gradle.kts

```diff
@@ -6,14 +9,29 @@ repositories {
     mavenCentral()
 }
 
+kotlin {
+    jvmToolchain(libs.versions.java.compilation.get().toInt())
+}
+
+// TODO: Remove setting `options.release` and `compilerOptions.jvmTarget` after upgrade to Kotlin Gradle Plugin 1.9
+//  it would be OK if `build-logic` produced Java 20 compatible bytecode, but setting target to Java 20 is supported only in KGP 1.9
+//  @see https://github.com/pinterest/ktlint/pull/2120#discussion_r1260229055 for more details
+val targetJavaVersion = JavaVersion.VERSION_17
```

Comment:
The name `targetJavaVersion` is now confusing compared to `java-target` in `gradle/libs.versions.toml`.

Reply:
I renamed the variable to `buildLogicTargetJavaVersion`, I hope that's fine for now (+ I hope this code will be removed soon™, after KGP 1.9 upgrade)